tests/models/vocos/test_modeling_vocos.py

```diff
@@ -0,0 +1,630 @@
+                    msg += str(e)
+                    raise AssertionError(msg)
+
+        set_model_tester_for_less_flaky_test(self)
+
+        config, batched_input = self.model_tester.prepare_config_and_inputs_for_common()
+        set_config_for_less_flaky_test(config)
+
+        for model_class in self.all_model_classes:
+            config.output_hidden_states = True
+
+            model_name = model_class.__name__
+            if hasattr(self.model_tester, "prepare_config_and_inputs_for_model_class"):
+                config, batched_input = self.model_tester.prepare_config_and_inputs_for_model_class(model_class)
+            batched_input_prepared = self._prepare_for_class(batched_input, model_class)
+            model = model_class(config).to(torch_device).eval()
+            set_model_for_less_flaky_test(model)
+
+            batch_size = self.model_tester.batch_size
+            single_row_input = {}
+            for key, value in batched_input_prepared.items():
+                if isinstance(value, torch.Tensor) and value.dim() > 0 and value.shape[0] % batch_size == 0:
+                    # e.g. musicgen has inputs of size (bs*codebooks). in most cases value.shape[0] == batch_size
+                    single_batch_shape = value.shape[0] // batch_size
+                    single_row_input[key] = value[:single_batch_shape]
+                else:
+                    single_row_input[key] = value
+
+            with torch.no_grad():
+                model_batched_output = model(**batched_input_prepared)
+                model_row_output = model(**single_row_input)
+
+            if isinstance(model_batched_output, torch.Tensor):
+                model_batched_output = {"model_output": model_batched_output}
+                model_row_output = {"model_output": model_row_output}
+
+            for key in model_batched_output:
+                # DETR starts from zero-init queries to decoder, leading to cos_similarity = `nan`
+                if hasattr(self, "zero_init_hidden_state") and "decoder_hidden_states" in key:
+                    model_batched_output[key] = model_batched_output[key][1:]
+                    model_row_output[key] = model_row_output[key][1:]
+                recursive_check(model_batched_output[key], model_row_output[key], model_name, key)
+
+
+@require_torch
+class VocosModelIntegrationTest(unittest.TestCase):
+    def setUp(self):
+        self.feature_extractor = VocosFeatureExtractor.from_pretrained("Manel/Vocos", resume_download=True)
+        self.model = VocosModel.from_pretrained("Manel/Vocos", resume_download=True).to(torch_device).eval()
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+        ds = ds.cast_column("audio", Audio(sampling_rate=self.feature_extractor.sampling_rate))
+        self.speech = ds[0]["audio"]["array"]
+
+    def test_inference(self):
+        EXPECTED = torch.tensor(
```

Comment:
could you add link to reproducer for these expected outputs? Like [here](https://github.com/huggingface/transformers/blob/0a21e870c76262eb54cf6d7fde0f6c22b962204c/tests/models/xcodec/test_modeling_xcodec.py#L430). It will help for reproducibility and future debugging 🙂